src/lib/scheduler/abstract_task.hpp

```diff
@@ -110,6 +110,11 @@ class AbstractTask : public std::enable_shared_from_this<AbstractTask> {
    */
   void execute();
 
+  /**
+   * What an awful name.
+   */
+  std::atomic_bool about_to_be_executed{false};
```

Comment:
Why not `to_be_executed`?

Reply:
If a task isn't "to be executed" - what is it there for?

`execution_imminent`? :D

Reply:
I'll change it to proper state transitions: Created->Scheduled->Enqueued->Dequeued->Running->Completed

Reply:
There is an open review, but I am currently waiting for the renaming. Does that make sense or do you want me to look at something in particular?

Reply:
I renamed it to `_is_assigned_to_worker`. Rewriting this into proper state transitions is still something that should be done, but brings too many changes with it to be done as a side thing in this PR. :/